plan/physical_plan_builder.go

```diff
@@ -783,6 +785,261 @@ func (p *Union) convert2PhysicalPlan(prop *requiredProperty) (*physicalPlanInfo,
 	return info, nil
 }
 
+func buildTableScanByKeyAndCorCol(p *DataSource, pkName model.CIStr, fakeConds []expression.Expression, origConds []expression.Expression) (*physicalPlanInfo, []expression.Expression, []expression.Expression) {
```

Comment:
I seems we don't need to do like that. Building filter condition during runtime is more reasonable.